jdaviz/configs/specviz/plugins/line_analysis/line_analysis.py

```diff
@@ -332,7 +332,31 @@ def _uncertainty(result):
         for function in FUNCTIONS:
             # TODO: update specutils to allow ALL analysis to take regions and continuum so we
             # don't need these if statements
-            if function == "Equivalent Width":
+            if function == "Line Flux":
+                meta = self.dataset.selected_dc_item.meta
+                telescope = meta.get('TELESCOP',
+                                     meta.get('telescope',
+                                              meta.get('_primary_header', {}).get('TELESCOP', '')))
+                if telescope == 'JWST':
+                    # Perform integration in frequency space
+                    freq_spec = Spectrum1D(
+                        spectral_axis=spec_subtracted.spectral_axis.to(u.Hz,
+                                                                       equivalencies=u.spectral()),
+                        flux=spec_subtracted.flux)
+                    temp_result = analysis.line_flux(freq_spec)
```

Comment:
Is there a reason this logic is isolated to JWST?  Is it just because we know that the input units will be compatible with the desired output units?  If so, can we check for unit compatibility instead?

And I still think it would be useful to check if specutils returns a different answer depending on whether the computation is done in frequency or wavelength.  If it isn't different, then these lines aren't necessary and can save from having to create a new Spectrum1D object on the fly.  But if it is different, then we may want to either move this logic upstream, or at least add a warning and perhaps an optional switch in specutils to force the computation to be done in frequency space.  That could be a follow-up effort if we want to get this quick patch in, we'll just have to make a note to come back and remove the exception here if we do make changes in specutils.

Reply:
@kecnry @PatrickOgle and I talked about this offband that I'm going to update here:

1) After the scope change, this logic should not be limited to just JWST. I've removed the jwst logic and just converted it to W/m^2 if the line flux result is in the right units
2) We're going to defer the specutils investigation to a later date. I'll make a new ticket to make sure we don't forget

Reply:
As for the second point, https://docs.astropy.org/en/stable/units/equivalencies.html#spectral-flux-and-luminosity-density-units might be useful:

> There is also support for spectral flux and luminosity density units, their equivalent surface brightness units, and integrated flux units. Their use is more complex, since it is necessary to also supply the location in the spectrum for which the conversions will be done, and the units of those spectral locations. The function that handles these unit conversions is [spectral_density()](https://docs.astropy.org/en/stable/api/astropy.units.equivalencies.spectral_density.html#astropy.units.equivalencies.spectral_density).